### DISCUSSION TITLE
Check if a file exists when used as a framework.

### DISCUSSION THREAD

**Discussion by jonshipmansmc** [pocketbase/pocketbase#3996] [Q&A]
Using `$os.writeFile` as part of a `cronAdd` callback. Is there framework/jsvm way to check for the existence of a file? `$os.remove` will not fail silently even when wrapped in a try...catch.

**Comment by ganigeorgiev** [comment]
> $os.remove will not fail silently even when wrapped in a try...catch

It works fine for me:

```js
try {
    $os.remove("something_that_does_not_exists");
} catch (err) {
    console.log("CATCH:", err)
}
```
Output:
```sh
2023/12/19 16:55:41 CATCH: GoError: remove something_that_does_not_exists: no such file or directory
```

**Reply by jonshipmansmc** [reply]
By george, I think you're onto something. I think there must have been something wrong somewhere else in the file because now it works fine.

Still having the original problem checking if the file exists. To get into it a little more, This is what I'm doing:

```
const writeFile = (url, filename) => {
	if (!url) return {};
	const dir = $os.tempDir();
	let file = '';
	let filepaths = [];

	const response = $http.send({
		url,
		timeout: 500
	});

	if (response?.raw && response.statusCode < 399) {
		const tmpfile = $filepath.join(dir, filename);
		const mp3file = $filepath.join(dir, filename + '.mp3');

		try {
			$os.remove(tmpfile);
		} catch (e) {
			// noop.
		}

		$os.writeFile(tmpfile, response.raw);

		$os.cmd('ffmpeg', '-i', tmpfile, '-f', 'mp3', '-b:a', '96k', mp3file).output();

		file = $filesystem.fileFromPath(mp3file);
		filepaths.push(tmpfile, mp3file);
	}

	return { file, filepaths };
};
```

What I want, is if the ffmpeg fails, to use the `tmpfile` instead of the `mp3file`. `ffmpeg` (at least on Windows) doesn't seem to throw errors on failures as it will return an empty output. So my thought was to check for the file after the $os.cmd('ffmpeg' line and use either the mp3 or tmp file. Would also cover cases when ffmpeg is unavailable.

**Reply by jonshipmansmc** [reply]
I've found this works - though it is to be seen how performant it is

```
let fileExists = false;
try {
	fileExists = !!$os.readFile(mp3file);
} catch (e) {
	// noop.
}
```
